lib/Controller/AppointmentConfigController.php

```diff
@@ -0,0 +1,257 @@
+ *
+ */
+
+namespace OCA\Calendar\Controller;
+
+use OCA\Calendar\Exception\ClientException;
+use OCA\Calendar\Exception\ServiceException;
+use OCA\Calendar\Http\JsonResponse;
+use OCA\Calendar\Service\Appointments\AppointmentConfigService;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http;
+use OCP\IRequest;
+use Psr\Log\LoggerInterface;
+
+class AppointmentConfigController extends Controller {
+
+	/** @var AppointmentConfigService */
+	private $appointmentConfigService;
+
+	/** @var string|null */
+	private $userId;
+
+	/** @var LoggerInterface */
+	private $logger;
+
+	public function __construct(string $appName,
+								IRequest $request,
+								AppointmentConfigService $appointmentService,
+								LoggerInterface $logger,
+								?string $userId) {
+		parent::__construct($appName, $request);
+		$this->appointmentConfigService = $appointmentService;
+		$this->userId = $userId;
+		$this->logger = $logger;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 */
+	public function index(): JsonResponse {
+		if ($this->userId === null) {
+			$this->logger->warning('No user found');
+			return JsonResponse::fail(null, Http::STATUS_NOT_FOUND);
+		}
+
+		try {
+			$appointmentConfigs = $this->appointmentConfigService->getAllAppointmentConfigurations($this->userId);
+			return JsonResponse::success($appointmentConfigs);
+		} catch (ServiceException $e) {
+			$this->logger->error('No appointment configurations found', ['exception' => $e]);
+			return JsonResponse::fail([], $e->getHttpCode() ?? Http::STATUS_BAD_REQUEST);
+		}
+	}
+
+	/**
+	 * @param int $id
+	 *
+	 * @return JsonResponse
+	 */
+	public function show(int $id): JsonResponse {
+		if ($this->userId === null) {
+			$this->logger->warning('No user found');
+			return JsonResponse::fail();
+		}
+
+		try {
+			$appointmentConfig = $this->appointmentConfigService->findByIdAndUser($id, $this->userId);
+			return JsonResponse::success($appointmentConfig);
+		} catch (ClientException $e) {
+			$this->logger->warning('No appointment configuration found with id ' . $id, ['exception' => $e]);
+			return JsonResponse::fail([], $e->getHttpCode() ?? Http::STATUS_BAD_REQUEST);
+		}
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @param string $name
+	 * @param string $description
+	 * @param string $location
+	 * @param string $visibility
+	 * @param string $targetCalendarUri
+	 * @param string|null $availability
+	 * @param int $length
+	 * @param int $increment
+	 * @param int $preparationDuration
+	 * @param int $followupDuration
+	 * @param int $timeToNextSlot
+	 * @param int|null $dailyMax
+	 * @param string[]|null $freebusyUris
+	 * @param int|null $start
+	 * @param int|null $end
+	 * @return JsonResponse
+	 */
+	public function create(
+		string $name,
+		string $description,
+		string $location,
+		string $visibility,
+		string $targetCalendarUri,
+		?string $availability,
+		int $length,
+		int $increment,
+		int $preparationDuration = 0,
+		int $followupDuration = 0,
+		int $timeToNextSlot = 0,
```

Comment:
This field is called `timeBeforeToNextSlot` on the model. We should decide on one of the two names.